blockchain/blockchain.go

```diff
@@ -449,14 +449,27 @@ func stateUpdateByHash(txn db.Transaction, hash *felt.Felt) (*core.StateUpdate,
 }
 
 // SanityCheckNewHeight checks integrity of a block and resulting state update
-func (b *Blockchain) SanityCheckNewHeight(block *core.Block, stateUpdate *core.StateUpdate) error {
+func (b *Blockchain) SanityCheckNewHeight(block *core.Block, stateUpdate *core.StateUpdate, classes map[felt.Felt]core.Class,
+) []error {
+	var vErrs []error
```

Comment:
It doesn't make sense to wrap errors and then also have a list of errors. We should do one or the other.

Reply:
Could we then log errors here like you recommended?